### PR TITLE
policy:  Add new constant MAX_STANDARD_MULTISIG_KEYS

### PR DESCRIPTION
**Pull request by dajohi**

### CODE REVIEW

File: src/policy/policy.cpp [hunk 1]
```diff
@@ -44,8 +44,9 @@ bool IsStandard(const CScript& scriptPubKey, txnouttype& whichType)
     {
         unsigned char m = vSolutions.front()[0];
         unsigned char n = vSolutions.back()[0];
-        // Support up to x-of-3 multisig txns as standard
-        if (n < 1 || n > 3)
+        // Support up to x-of-MAX_STANDARD_MULTISIG_KEYS multisig txns as
+        // standard
```
Comment (dcousens) [inline src/policy/policy.cpp]: `Support up to m-of-MAX_ ...`

What is this `x`?

**Comment by dcousens** [comment]
utACK

**Comment by dajohi** [comment]
@dcousens Agree.  Updated.

**Comment by laanwj** [comment]
Trivial ACK

**Comment by gmaxwell** [comment]
I think the result of this is confusing, if you go read policy.h post patch you think that bitcoin multisig will not allow more keys. But this isn't true, the restriction is specific to bare multisig-- something virtually no one cares about-- which was more or less apparent from the code in context but isn't otherwise; and in the updated version the new comment is arguably outright factually incorrect.

FWIW, the constant movement of trivial bits of code behavior across multiple files is very frustrating to me.  The net result of this is that I am less likely to review future changes that changes around that test in IsStandard, because I have to go scatter and read many different files (which involves first figuring out _where_ its defined) to have any idea of what the code actually does.

**Comment by luke-jr** [comment]
IMO this should become an option or bare multisig made unconditionally non-IsStandard.

**Comment by sipa** [comment]
@gmaxwell Seems that at least the comment should be fixed to say it's about bare multisig.

File: src/policy/policy.h [hunk 2]
```diff
@@ -26,6 +26,12 @@ static const unsigned int MAX_P2SH_SIGOPS = 15;
 /** The maximum number of sigops we're willing to relay/mine in a single tx */
 static const unsigned int MAX_STANDARD_TX_SIGOPS = MAX_BLOCK_SIGOPS/5;
 /**
+ * The maximum number of public keys allowed in a multi-signature transaction
+ * output script to be considered standard.
+ */
+static const unsigned int MAX_STANDARD_MULTISIG_KEYS = 3;
```
Comment (dcousens) [inline src/policy/policy.h]: Maybe `MAX_STANDARD_BARE_MULTISIG_KEYS`, in light of @gmaxwell's comment

**Comment by dcousens** [comment]
Agreed the indirection of extracting constants can be frustrating for review, but, IMHO the only [equally balanced] alternative is to remove the constants themselves.

This is probably an exception though, given it is a 1:1 mapping and the motivation for extraction is solely a style choice.   I think the underlying question should be: if we are encouraging users to modify these constants (by extracting them all to 1 place),  should we be pointing them at tweaking constants or tweaking the policy code itself.

**Comment by laanwj** [comment]
Too controversial for a trivial change, closing

**Pull request closed without merging**